### PR TITLE
DataIntegrityChecksJob: detect multiple open root-child tasks

### PR DESCRIPTION
**Pull request by yoomlam**
Warns in Slack about bad task tree with multiple open root-children tasks: https://github.com/department-of-veterans-affairs/dsva-vacols/issues/247#issuecomment-972929235

Stacked on top of #17015.

### Description
There should be no more than 1 open among these root-children tasks: 
`DistributionTask JudgeAssignTask JudgeDecisionReviewTask QualityReviewTask BvaDispatchTask`
Add `MultipleOpenRootChildTaskChecker` to `DataIntegrityChecksJob` to warn about such appeals.

[Investigation of offending appeals](https://dsva.slack.com/archives/CQTDX9BF0/p1638482890034400)

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Warns in Slack when more than 1 open among the root-children tasks for each appeal

### Testing Plan
See RSpec

### CODE REVIEW

File: app/services/multiple_open_root_child_task_checker.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "#appeals-echo"
+  end
+
+  def self.open_exclusive_root_children_tasks(appeal)
+    appeal.tasks.open.of_type(EXCLUSIVE_OPEN_TASKS).where(parent: appeal.root_task)
+  end
+
+  private
+
+  def appeals_with_multiple_open_root_child_task
```
Comment (eileen-nava) [inline app/services/multiple_open_root_child_task_checker.rb:26]: (This isn't a blocking comment.  I'm asking it for my own curiosity.)
Why did you make this method private?

File: app/services/multiple_open_root_child_task_checker.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  def open_root_tasks
+    RootTask.open
+  end
+
+  def build_report(appeals)
+    return if appeals.empty?
+
+    count = appeals.count
+    add_to_report "Found #{count} #{'appeal'.pluralize(count)} with multiple open root-children task types:"
```
Comment (eileen-nava) [inline app/services/multiple_open_root_child_task_checker.rb:41]: This is the first I've seen pluralize.  TIL!

**Review by eileen-nava — APPROVED** [review]
Looks good. 👍

File: app/services/multiple_open_root_child_task_checker.rb (hunk 1 above)
Comment (yoomlam) [reply app/services/multiple_open_root_child_task_checker.rb:26]: B/c it was not designed to be used outside of this class. Coming from an OO background, my inclination is methods should be private by default unless they're designed to be used outside the class. Your question led me to https://bootrails.com/blog/ruby-private-method.

File: app/services/multiple_open_root_child_task_checker.rb (hunk 2 above)
Comment (yoomlam) [reply app/services/multiple_open_root_child_task_checker.rb:41]: It's a nice touch but I wish the syntax was shorter.

File: app/services/multiple_open_root_child_task_checker.rb (hunk 1 above)
Comment (eileen-nava) [reply app/services/multiple_open_root_child_task_checker.rb:26]: Got it, thanks.

**Pull request merged**